src/unix.rs

```diff
@@ -258,6 +280,65 @@ impl Client {
         }
     }
 
+    pub fn try_acquire(&self) -> io::Result<Option<Acquired>> {
+        let mut buf = [0];
+
+        #[cfg(target_os = "linux")]
+        {
+            let read = self.read().as_raw_fd();
+            loop {
+                match non_blocking_read(read, &mut buf) {
+                    Ok(1) => return Ok(Some(Acquired { byte: buf[0] })),
+                    Ok(_) => {
+                        return Err(io::Error::new(
+                            io::ErrorKind::UnexpectedEof,
+                            "early EOF on jobserver pipe",
+                        ))
+                    }
+
+                    Err(e) if e.kind() == io::ErrorKind::WouldBlock => return Ok(None),
+                    Err(e) if e.kind() == io::ErrorKind::Interrupted => continue,
+                    Err(e) if e.kind() == io::ErrorKind::Unsupported => break,
+
+                    Err(err) => return Err(err),
+                }
+            }
+        }
+
+        let (mut fifo, is_non_blocking) = if let Self::Fifo {
+            file,
+            is_non_blocking,
+            ..
+        } = self
+        {
+            (file, is_non_blocking)
```

Comment:
This looks terrible. Imo just using a match and having some rightwards drift would be better here. Or let-else if the MSRV allows.

Reply:
Our msrv is 1.63 and let-else is stablised in 1.65, so I've replaced it with `match`